specification/hardwaresecuritymodules/HardwareSecurityModules.Management/CloudHsmCluster.tsp

```diff
@@ -0,0 +1,240 @@
+   */
+  listBySubscription is ArmListBySubscription<
+    CloudHsmCluster,
+    Parameters = {
+      /**
+       * The page-continuation token to use with a paged version of this API
+       */
+      @query("$skiptoken")
+      $skiptoken?: string;
+    }
+  >;
+
+  /**
+   * Pre Backup operation to validate whether the customer can perform a backup on the Cloud HSM Cluster resource in the specified subscription.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header"
+  validateBackupProperties is ArmResourceActionAsync<
+    CloudHsmCluster,
+    BackupRequestProperties,
+    ArmResponse<BackupResult>,
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = BackupResult> &
+      Azure.Core.Foundations.RetryAfterHeader,
+    OptionalRequestBody = true
+  >;
+
+  /**
+   * Create a backup of the Cloud HSM Cluster in the specified subscription
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header"
+  backup is ArmResourceActionAsync<
+    CloudHsmCluster,
+    BackupRequestProperties,
+    ArmResponse<BackupResult>,
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = BackupResult> &
+      Azure.Core.Foundations.RetryAfterHeader,
+    OptionalRequestBody = true
+  >;
+
+  /**
+   * Queued validating pre restore operation
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header"
+  validateRestoreProperties is ArmResourceActionAsync<
+    CloudHsmCluster,
+    RestoreRequestProperties,
+    ArmResponse<RestoreResult>,
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = RestoreResult> &
+      Azure.Core.Foundations.RetryAfterHeader,
+    OptionalRequestBody = true
+  >;
+
+  /**
+   * Restores all key materials of a specified Cloud HSM Cluster
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header"
+  restore is ArmResourceActionAsync<
+    CloudHsmCluster,
+    RestoreRequestProperties,
+    ArmResponse<RestoreResult>,
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = RestoreResult> &
+      Azure.Core.Foundations.RetryAfterHeader
+  >;
+
+  /**
+   * Gets the private link resources supported for the Cloud Hsm Cluster.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @get
+  @action("privateLinkResources")
+  @operationId("CloudHsmClusterPrivateLinkResources_ListByCloudHsmCluster")
+  listByCloudHsmCluster is ArmResourceActionSync<
+    CloudHsmCluster,
+    void,
+    CommonTypes.PrivateLinkResourceListResult
+  >;
+
+  /**
+   * Gets the backup operation status of the specified Cloud HSM Cluster
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-operation"
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header"
+  @operationId("CloudHsmClusterBackupStatus_Get")
+  @get
+  cloudHsmClusterBackupStatusGet(
```

Comment:
Same question and it applies to other operations that needed the operationId decorator, why aren't they in an interface that corresponds to the name in the prefix of the operationId? In this case I would expect this operation to be in a CloudHsmClusterBackupStatus interface. Mostly calling this out because it may change the way operations are generated downstream in other language emitters unless there were transforms

Reply:
Because we group operations by resource, not by operation group. The downstream issue will be handled by https://github.com/Azure/typespec-azure/issues/2683